docs/arguments.md

```diff
@@ -25,18 +23,30 @@ new Arguments(new { logLevel = LogLevel.High });
 When you don't care about names of the dependencies you can pass them as array, in which case they will be matched by their type.
 
 ```csharp
-new Arguments(new[] { LogLevel.High });
+var args = new Arguments().InsertTyped(LogLevel.High);
 ```
 
 :information_source: **Typed arguments are matched exactly:** When you don't specify type of the dependency it's exact type will be used. So if you pass for example `MemoryStream` Windsor will try to match it to dependency of type `MemoryStream`, but not of the base type `Stream`. If you want to match it to `Stream` use `Insert` extension method.
 
 #### Custom dictionary of arguments
 
-You can also pass already populated dictionary to `Arguments` in which case its values will be copied over.
+You can also pass a dictionary to `Arguments` where it will be shallow copied.
+
+```csharp
+var args = new Arguments().Insert(new Dictionary<string, string> { { "anyString", "anyStringValue" } });
+```
 
-### `Insert` extension method
+#### Custom read-only dictionary of arguments
 
-In addition to `Arguments` class there's also `Insert` extension method that extends `IDictionary`. It has several overloads that let you fluently insert values into the dictionary.
+You can also pass a dictionary to `Arguments` via the `InsertNamed` method which supports `IReadOnlyDictionary<string, object>` and is also shallow copied.
+
+```csharp
+var args = new Arguments().InsertNamed(new Dictionary<string, string> { { "anyString", "anyStringValue" } });
+```
+
+### `Insert` method
+
+In addition to `Arguments` class there's also an `Insert` method that extends `IDictionary`. It has several overloads that let you fluently insert values into the dictionary.
```

Comment:
Needs to be updated since it's no longer an extension method extending `IDictionary`?

Reply:
Correct, I did scan the docs for `extension` and managed to get those out. This needs re-writing.